Lib/webbrowser.py

```
@@ -484,6 +485,12 @@ def register_X_browsers():
 
 # Prefer X browsers if present
 if os.environ.get("DISPLAY"):
+    try:
+        cmd = "xdg-settings get default-web-browser".split()
+        _preferred_browser = subprocess.check_output(cmd).decode().strip()
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        pass
```

I am not sure you should `pass` the exceptions silently.

---

I'm not sure what to report, otherwise. Passing on the FileNotFoundError feels right to me.

I could drop catching CalledProcessError?

---

You could use shutil.which to determine if xdg-settings exists before trying to call it (LBYL).  But here ignoring the exception (EAFP) seems right to me.

---

With a `_preferred_browsers` set, this would become:

    try:
        cmd = "xdg-settings get default-web-browser".split()
        _xdg_default_browser = subprocess.check_output(cmd).decode().strip()
    except (FileNotFoundError, subprocess.CalledProcessError):
         pass
    else:
        _preferred_browsers.add(_xdg_default_browser)

I believe that would be more self-explanatory as to the purpose of ignoring the exceptions here (if we fail to find the xdg default browser, we simply don't add it to the preferred browser set)